frame/dapps-staking/src/lib.rs

```diff
@@ -502,6 +506,51 @@ impl Default for RewardDestination {
     }
 }
 
+#[derive(Encode, Decode, Clone, Copy, Default, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum EmbededDestination<AccountId, Balance: AtLeast32BitUnsigned + Default + Copy> {
+    #[default]
+    None,
+    Destination {
+        who: AccountId,
+        amount: Balance,
```

Comment:
It'd be good to have docs under enum values since for a first time reader, it's unclear to me why you need `amount` attribute (probably will be clear as I continue reading on).

Reply:
I apologise for this. I must've skipped this.
I saw that there are enum types for indicating where the rewards are been sent to. i.e the stake balance, freebalance. 

So I added a new type for indicating the new destination which is the beneficiary delegated to receive the rewards.

Reply:
I see! (and no worries)

Perhaps just extending the existing enum with `DelegateReward` (and some extra logic to existing extrinsic calls) might do the trick then.

Reply:
I did, I just quoted the code now.